app/client/src/api/ApiUtils.ts

```diff
@@ -42,6 +42,7 @@ export const apiRequestInterceptor = (config: AxiosRequestConfig) => {
   const branch =
     getCurrentGitBranch(store.getState()) || getQueryParamsObject().branch;
   if (branch) {
+    // @ts-expect-error: config.headers might be undefined
     config.headers.branchName = branch;
```

Comment:
@yaldram Can we check for `config.headers` as well here? 
```suggestion
    if(branch && config.headers)
        config.headers.branchName = branch;
```
Because, if `config.headers` is actually `undefined`, this seems like it will throw an error. The alternative is to use `set` from `lodash`. Would you mind checking with the team member who added this `if(branch)` clause?

Reply:
I have added a check for the `config.headers`, because `AxiosRequestConfig` has all keys optional